src/openssl/ssl/context.cr

```diff
@@ -1,40 +1,121 @@
+      super(method)
+    end
+  end
+
+  class Server < Context
+    # Generates a new SSL server context with sane defaults for a server connection.
+    #
+    # By default it defaults to the `SSLv23_method` which actually means that
+    # OpenSSL will negotiate the TLS or SSL protocol to use with the remote
+    # endpoint.
+    #
+    # Don't change the method unless you must restrict a specific protocol to be
+    # used (eg: TLSv1.2) and nothing else. You should specify options to disable
+    # specific protocols, yet allow to negotiate from various other ones. For
+    # example the following snippet will enable the TLSv1, TLSv1.1 and TLSv1.2
+    # protocols but disable the deprecated SSLv2 and SSLv3 protocols:
+    #
+    # ```
+    # ssl_context = OpenSSL::SSL::Context::Server.new
+    # ssl_context.options = LibSSL::Options::NO_SSLV2 | LibSSL::Options::NO_SSLV3
+    # ```
+    def initialize(method : LibSSL::SSLMethod = LibSSL.sslv23_method)
+      super(method)
+      set_default_verify_paths
+    end
+
+    # Returns a new SSL server context with only the given method set.
+    #
+    # For everything else this uses the defaults of your OpenSSL.
+    # Use this only if undoing the defaults that `new` sets is too much hassle.
+    def self.insecure(method : LibSSL::SSLMethod = LibSSL.sslv23_method)
+      super(method)
+    end
+  end
+
+  protected def initialize(method : LibSSL::SSLMethod)
+    @handle = LibSSL.ssl_ctx_new(method)
+    raise OpenSSL::Error.new("SSL_CTX_new") if @handle.null?
   end
 
-  # Generates a new SSL context.
-  #
-  # By default it defaults to the `SSLv23_method` which actually means that
-  # OpenSSL will negotiate the TLS or SSL protocol to use with the remote
-  # endpoint.
-  #
-  # Don't change the method unless you must restrict a specific protocol to be
-  # used (eg: TLSv1.2) and nothing else. You should specify options to disable
-  # specific protocols, yet allow to negotiate from various other ones. For
-  # example the following snippet will enable the TLSv1, TLSv1.1 and TLSv1.2
-  # protocols but disable the deprecated SSLv2 and SSLv3 protocols:
-  #
-  # ```
-  # ssl_context = OpenSSL::SSL::Context.new
-  # ssl_context.options = LibSSL::Options::NO_SSLV2 | LibSSL::Options::NO_SSLV3
-  # ```
-  def initialize(method : LibSSL::SSLMethod = LibSSL.sslv23_method)
+  # Overriding initialize or new in the child classes as public methods,
+  # makes it either impossible to access the parent versions or makes the parent
+  # versions public too. So to provide insecure in the child classes, we need
+  # a second constructor that we call from there without getting the
+  # overridden ones of the childs.
+  protected def _initialize_insecure(method : LibSSL::SSLMethod)
     @handle = LibSSL.ssl_ctx_new(method)
     raise OpenSSL::Error.new("SSL_CTX_new") if @handle.null?
   end
 
+  protected def self.insecure(method : LibSSL::SSLMethod)
+    obj = allocate
```

Comment:
Hm, yeah, we'd need to duplicate that initialize in the subclasses. I tried that but I think I found a bug in the compiler, let's keep it like this for now :-)